src/api/core/queries/actionQuery.ts

```diff
@@ -0,0 +1,63 @@
+import { RequestExecuter } from "../../../internal/executer";
+import { ResourceType } from "../resource";
+import { QueryActionType } from "./../../../internal/utils";
+import { IFilteringCriterion, IFilteringCriterionCallback } from "./../../dataops/filteringApi";
+import { QueryAction } from "./baseQuery";
+import { FilterableQuery } from "./filterableQuery";
+
+/**
+ * Query object specialized for generic action statements.
+ * This object is generated automatically from the Dataset object, never to be instantiated directly.
+ *
+ * ### Example
+ * ```typescript
+ * // Attaching resources
+ * dataModule.dataset("posts")
+ *  .attach("comments", field => field("post.id").isEqualTo(somePostId))
+ *  .execute();
+ * ```
+ *
+ * @template T Generic type of your resource, default to any
+ */
+export class ActionQuery<T> extends FilterableQuery<T> {
+
+  /**
+   * @internal
+   */
+  public static create<T>(
```

Comment:
Is there any reason to have this static method as well as a private setQueryAction() method? You could've replaced both of them by just one line added to the constructor:
```
constructor(...) {
  super(queryExecuter, QueryAction.update, resourceType, resourceName);
  this.query.setAction(queryActionType, actionResourceName, filter);   
}
```

Reply:
perfect :-D